src/main/java/io/cdap/plugin/http/sink/batch/HTTPRecordWriter.java

```diff
@@ -62,17 +74,33 @@ public class HTTPRecordWriter extends RecordWriter<StructuredRecord, StructuredR
   private final HTTPSinkConfig config;
   private StringBuilder messages = new StringBuilder();
   private String contentType;
+  private String url;
+  private String configURL;
+  private List<PlaceholderBean> placeHolderList;
 
   private AccessToken accessToken;
+  private final HttpErrorHandler httpErrorHandler;
+  private final PollInterval pollInterval;
+  private HttpResponse response;
+  private int httpStatusCode;
 
   HTTPRecordWriter(HTTPSinkConfig config) {
     this.config = config;
     this.accessToken = null;
+    this.httpErrorHandler = new HttpErrorHandler(config);
+    if (config.getRetryPolicy().equals(RetryPolicy.LINEAR)) {
+      pollInterval = FixedPollInterval.fixed(config.getLinearRetryInterval(), TimeUnit.SECONDS);
+    } else {
+      pollInterval = IterativePollInterval.iterative(duration -> duration.multiply(2));
```

Comment:
As per [docs](https://github.com/awaitility/awaitility/wiki/Usage-(Legacy)#iterative-poll-interval) `Iterative Poll Interval` needs a start duration, here .multiply(2) will happen on 0, resulting all calls to be made instantaneously.

Reply:
I've added _500ms_ as the start duration. 4e4ee6211be03c9e368a811866c8d96ec49518d3